python/pyspark/errors/exceptions/captured.py

```diff
@@ -170,8 +170,10 @@ def getQueryContext(self) -> List[BaseQueryContext]:
         assert SparkContext._gateway is not None
 
         gw = SparkContext._gateway
-        if self._origin is not None and is_instance_of(
-            gw, self._origin, "org.apache.spark.SparkThrowable"
+        if (
+            self._origin is not None
+            and is_instance_of(gw, self._origin, "org.apache.spark.SparkThrowable")
+            and hasattr(self._origin, "getQueryContext")
```

Comment:
> So if getQueryContext is not explicitly overridden, it throws an exception

Interesting. In that case, do we also need to check other `default` methods, `getCondition`, `getMessageParameters`, and `getSqlState`?

Reply:
oh, actually `getCondition` is not a `default` method.